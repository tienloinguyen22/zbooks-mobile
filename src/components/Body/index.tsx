import { Body } from 'native-base';

export { Body };
