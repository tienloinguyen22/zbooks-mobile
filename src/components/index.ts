export * from './Body';
export * from './Button';
export * from './Card';
export * from './CardItem';
export * from './Container';
export * from './Content';
export * from './DatePicker';
export * from './ErrorBoundary';
export * from './ErrorText';
export * from './Field';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Image';
export * from './Input';
export * from './Item';
export * from './Label';
export * from './Left';
export * from './List';
export * from './ListItem';
export * from './Loading';
export * from './Picker';
export * from './Right';
export * from './ScrollView';
export * from './Text';
export * from './Touchable';
export * from './Thumbnail';
export * from './Tooltip';
export * from './View';
export * from './WithStore';
export * from './WithScreen';
export * from './Alert';
export * from './Animation';
export * from './AlertContainer';
