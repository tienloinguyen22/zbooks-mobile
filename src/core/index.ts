export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
