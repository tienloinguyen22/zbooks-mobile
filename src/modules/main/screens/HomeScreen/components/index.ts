export * from './NavigationSample';
export * from './IconSample';
export * from './RematchSample';
export * from './NotificationSample';
export * from './CrashSample';
export * from './PickerSample';
export * from './AnalyticsSample';
export * from './TextStyleSample';
