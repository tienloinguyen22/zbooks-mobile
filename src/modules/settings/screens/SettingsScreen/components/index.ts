export * from './SettingMenuItem';
export * from './LogoutModal';
