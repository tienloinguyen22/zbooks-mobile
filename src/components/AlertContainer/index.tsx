import React from 'react';
import { ViewStyle } from 'react-native';
import Modal from 'react-native-modal';
import { combineStyles, colors, THEME_DARK } from '@app/core';
import { useTranslation } from 'react-i18next';
import { useTheme } from '@app/hooks';
import { Icon } from '../Icon';
import { Button } from '../Button';
import { View } from '../View';
import { styles } from './styles';
import { Text } from '../Text';

export type AlertType = 'SUCCESS' | 'WARNING' | 'ERROR' | 'INFO' | 'DEFAULT';

export interface AlertAction {
  title: string;
  onPress: () => void;
  outline?: boolean;
}

interface AlertProps {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose: () => void;
  closeable?: boolean;
}

export const AlertContainer = (props: AlertProps): JSX.Element => {
  const { primaryColor, theme } = useTheme();
  const { t } = useTranslation();

  const backgroundAlertStyle = {
    backgroundColor: theme === THEME_DARK ? colors.lightBlack : colors.white,
  };

  const messageTextStyle = {
    color: theme === THEME_DARK ? colors.white : colors.black,
  };

  const closeModal = (): void => {
    props.closeable !== false && props.onPressClose();
  };

  let titleContainerStyle = combineStyles<ViewStyle>(styles.titleDefault, {
    backgroundColor: primaryColor,
  });
  let title = t('dialog.info');
  let iconName = 'information-outline';

  switch (props.type) {
    case 'WARNING':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.orange,
      });
      title = t('dialog.warning');
      iconName = 'alert-circle-outline';
      break;
    case 'ERROR':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.red,
      });
      title = t('dialog.error');
      iconName = 'alert-outline';
      break;
    case 'SUCCESS':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.green,
      });
      title = t('dialog.success');
      iconName = 'check';
      break;
    default:
      break;
  }

  props.title && ({ title } = props);

  return (
    <View style={styles.container}>
      <Modal isVisible={true} onBackButtonPress={closeModal} onBackdropPress={closeModal}>
        <View style={[styles.alertContainer, backgroundAlertStyle]}>
          <View style={titleContainerStyle}>
            <Icon name={iconName} color={colors.white} size={25} />
            <Text style={styles.titleText}>{title}</Text>
            <View style={styles.dialogIcon} />
          </View>
          <Text style={[styles.messageText, messageTextStyle]}>{props.message}</Text>
          {props.actions && (
            <View style={styles.buttonContainer}>
              {props.actions.map((item) => {
                return (
                  <Button outline={item.outline} onPress={item.onPress} key={item.title} style={styles.buttonDialog}>
                    <Text white={!item.outline} primary={item.outline}>
                      {item.title}
                    </Text>
                  </Button>
                );
              })}
            </View>
          )}
          {!props.actions && (
            <View style={styles.reserveContainer}>
              <Button onPress={props.onPressClose} style={styles.buttonDialog}>
                <Text white>{t('dialog.close')}</Text>
              </Button>
            </View>
          )}
        </View>
      </Modal>
    </View>
  );
};
