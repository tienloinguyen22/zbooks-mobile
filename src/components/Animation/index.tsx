import React from 'react';
import { Resource } from '@app/core';
import LottieView from 'lottie-react-native';

interface Props {
  width: number;
  height: number;
  source: Resource;
  autoPlay?: boolean;
  loop?: boolean;
}

export const Animation = (props: Props): JSX.Element => {
  return (
    <LottieView
      source={props.source}
      style={{
        width: props.width,
        height: props.height,
      }}
      autoPlay={props.autoPlay === true}
      loop={props.loop === true}
    />
  );
};
