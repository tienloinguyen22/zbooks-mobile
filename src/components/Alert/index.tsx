import React from 'react';
import RootSiblings from 'react-native-root-siblings';
import { AlertContainer, AlertAction, AlertType } from '../AlertContainer';

interface ShowParams {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose?: () => void;
  closeable?: boolean;
}

let alert: RootSiblings | undefined;
const hide = (): void => {
  alert && alert.destroy();
};

const show = (props: ShowParams): RootSiblings => {
  alert && hide();
  alert = new RootSiblings(<AlertContainer {...props} onPressClose={props.onPressClose || hide} />);
  return alert;
};

export const Alert = {
  show,
  hide,
};
