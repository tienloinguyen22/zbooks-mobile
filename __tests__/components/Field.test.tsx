import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Field, Picker, PickerDataItem } from '@app/components';
import { THEME_DARK, THEME_LIGHT } from '@app/core';
import { mockTheme } from '../test_helpers';

describe('components/Field', () => {
  const labelText = 'label';
  const valueText = 'value';
  const errorText = 'error';
  const tooltipText = 'tooltip';
  beforeEach(() => {
    mockTheme(THEME_DARK);
  });

  it('renders successfully', async () => {
    const { baseElement } = render(<Field label={labelText} value={valueText} />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with light theme successfully', async () => {
    mockTheme(THEME_LIGHT);
    const { baseElement } = render(<Field label={labelText} value={valueText} />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with error successfully', async () => {
    const { baseElement, getByText } = render(
      <Field label={labelText} value={valueText} error errorMessage={errorText} />,
    );

    expect(getByText(errorText)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders with success successfully', async () => {
    const { baseElement } = render(<Field label={labelText} value={valueText} success />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with tooltip successfully', async () => {
    const { baseElement, getByText } = render(
      <Field
        label={labelText}
        value={valueText}
        hasTooltip
        tooltipHeight={100}
        tooltipWidth={100}
        tooltipText={tooltipText}
      />,
    );

    expect(getByText(tooltipText)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders with tooltip without specifying width and height successfully', async () => {
    const { baseElement, getByText } = render(
      <Field label={labelText} value={valueText} hasTooltip tooltipText={tooltipText} />,
    );

    expect(getByText(tooltipText)).toBeDefined();
    expect(baseElement).toMatchSnapshot();
  });

  it('changes text value successfully', async () => {
    const changedText = 'changed';

    const onChangeText = jest.fn();
    const { getByDisplayValue } = render(
      <Field label={labelText} value={valueText} type={'text'} onChangeText={onChangeText} />,
    );
    fireEvent.changeText(getByDisplayValue(valueText), changedText);

    expect(onChangeText).toBeCalledWith(changedText);
  });

  it('shows picker when pressing text (Picker Field)', async () => {
    Picker.show = jest.fn();

    const pickerData: PickerDataItem<string>[] = [
      {
        value: 'data1',
        text: 'text1',
      },
      {
        value: 'data2',
        text: 'text2',
      },
    ];
    const { getByText } = render(
      <Field label={labelText} value={'data1'} type={'picker'} pickerDataSources={pickerData} />,
    );
    fireEvent.press(getByText('text1'));

    expect(Picker.show).toBeCalledTimes(1);
  });

  it('does not show picker when pressing text (Picker Field) if dataSource is undefined', async () => {
    Picker.show = jest.fn();
    const { getByText } = render(<Field label={labelText} value={'data1'} type={'picker'} />);
    fireEvent.press(getByText(''));

    expect(Picker.show).not.toBeCalled();
  });

  it('changes value of Picker Field successfully', async () => {
    const mockShow: <T>(options: {
      dataSources: PickerDataItem<T>[];
      initialValue?: T;
      onValueChanged: (value: T, item: PickerDataItem<T>) => void;
    }) => void = (options) => {
      options.onValueChanged &&
        options.onValueChanged(
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          'data2' as any,
          {
            value: 'data1',
            text: 'text1',
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
          } as any,
        );
    };
    Picker.show = mockShow;
    const onChangeText = jest.fn();

    const pickerData: PickerDataItem<string>[] = [
      {
        value: 'data1',
        text: 'text1',
      },
      {
        value: 'data2',
        text: 'text2',
      },
    ];
    const { getByText } = render(
      <Field
        label={labelText}
        value={'data1'}
        type={'picker'}
        pickerDataSources={pickerData}
        onChangeText={onChangeText}
      />,
    );
    fireEvent.press(getByText('text1'));

    expect(onChangeText).toBeCalledTimes(1);
  });
});
