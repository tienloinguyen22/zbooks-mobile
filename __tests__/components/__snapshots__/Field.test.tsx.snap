// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Field renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 0,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "textTransform": "none",
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          label
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "borderRadius": 25,
              "borderWidth": 1,
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "borderColor": "black",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="label"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "white",
            }
          }
          underlineColorAndroid="transparent"
          value="value"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`components/Field renders with error successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 0,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "textTransform": "none",
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          label
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "borderRadius": 25,
              "borderWidth": 1,
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "borderColor": "#FC4173",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="label"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "white",
            }
          }
          underlineColorAndroid="transparent"
          value="value"
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "textTransform": "none",
            },
            Object {
              "color": "#FC4173",
              "fontSize": 13,
            },
            Object {
              "marginBottom": 5,
              "marginLeft": 15,
              "marginTop": 5,
            },
            Object {
              "color": "#FC4173",
            },
          ]
        }
      >
        error
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/Field renders with light theme successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 0,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "textTransform": "none",
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          label
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "borderRadius": 25,
              "borderWidth": 1,
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "borderColor": "black",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="label"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "white",
            }
          }
          underlineColorAndroid="transparent"
          value="value"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`components/Field renders with success successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 0,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "textTransform": "none",
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          label
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "borderRadius": 25,
              "borderWidth": 1,
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "borderColor": "#24DF98",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="label"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "white",
            }
          }
          underlineColorAndroid="transparent"
          value="value"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`components/Field renders with tooltip successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 0,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "textTransform": "none",
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          label
        </Text>
        <View
          collapsable={false}
        >
          <TouchableOpacity
            activeOpacity={1}
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                  },
                  Object {
                    "margin": 8,
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </TouchableOpacity>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <TouchableOpacity
              activeOpacity={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "height": 0,
                      "left": 0,
                      "overflow": "visible",
                      "position": "absolute",
                      "top": 0,
                      "width": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 14,
                        },
                        Object {
                          "margin": 8,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "left": -7.5,
                      "position": "absolute",
                      "top": -2,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderBottomColor": "white",
                          "borderBottomWidth": 15,
                          "borderLeftColor": "transparent",
                          "borderLeftWidth": 8,
                          "borderRightColor": "transparent",
                          "borderRightWidth": 8,
                          "borderStyle": "solid",
                          "height": 0,
                          "width": 0,
                        },
                        Object {
                          "borderBottomColor": "#fff",
                        },
                        Object {},
                      ]
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "display": "flex",
                      "flex": 1,
                      "height": 100,
                      "justifyContent": "flex-start",
                      "left": -17.999,
                      "padding": 10,
                      "position": "absolute",
                      "top": 10,
                      "width": 100,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#222C44",
                        "textTransform": "none",
                      }
                    }
                  >
                    tooltip
                  </Text>
                </View>
              </View>
            </TouchableOpacity>
          </Modal>
        </View>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "borderRadius": 25,
              "borderWidth": 1,
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "borderColor": "black",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="label"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "white",
            }
          }
          underlineColorAndroid="transparent"
          value="value"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`components/Field renders with tooltip without specifying width and height successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 0,
            "paddingTop": 10,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        center={true}
        row={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "textTransform": "none",
              },
              Object {
                "marginBottom": 5,
                "marginTop": 5,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          label
        </Text>
        <View
          collapsable={false}
        >
          <TouchableOpacity
            activeOpacity={1}
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                  },
                  Object {
                    "margin": 8,
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </TouchableOpacity>
          <Modal
            animationType="fade"
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <TouchableOpacity
              activeOpacity={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "flex": 1,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "height": 0,
                      "left": 0,
                      "overflow": "visible",
                      "position": "absolute",
                      "top": 0,
                      "width": 0,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 14,
                        },
                        Object {
                          "margin": 8,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "left": -7.5,
                      "position": "absolute",
                      "top": -2,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "borderBottomColor": "white",
                          "borderBottomWidth": 15,
                          "borderLeftColor": "transparent",
                          "borderLeftWidth": 8,
                          "borderRightColor": "transparent",
                          "borderRightWidth": 8,
                          "borderStyle": "solid",
                          "height": 0,
                          "width": 0,
                        },
                        Object {
                          "borderBottomColor": "#fff",
                        },
                        Object {},
                      ]
                    }
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "display": "flex",
                      "flex": 1,
                      "height": 50,
                      "justifyContent": "flex-start",
                      "left": -17.999,
                      "padding": 10,
                      "position": "absolute",
                      "top": 10,
                      "width": 250,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#222C44",
                        "textTransform": "none",
                      }
                    }
                  >
                    tooltip
                  </Text>
                </View>
              </View>
            </TouchableOpacity>
          </Modal>
        </View>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "borderRadius": 25,
              "borderWidth": 1,
              "fontSize": 15,
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "borderColor": "black",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          placeholder="label"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "white",
            }
          }
          underlineColorAndroid="transparent"
          value="value"
        />
      </View>
    </View>
  </View>
</View>
`;
